tenable/io/was/iterator.py

```diff
@@ -0,0 +1,133 @@
+from typing import Any, List, Dict
+from restfly import APIIterator
+from tenable.io.base import TIOIterator
+
+
+class WasIterator(APIIterator):
+    """
+    The WAS iterator can be used to handle downloading and processing of the data from
+    WAS export.
+
+    Attributes:
+        target_scan_ids:
+            The target Scan IDs for the specified parent Scan ID.
+        page (list[dict]):
+            The current chunk of data.
+    """
+    target_scan_ids: List[str]
+    page: List[Dict]
+
+    def __init__(self, api, target_scan_ids, **kwargs):
+        self.target_scan_ids = target_scan_ids
+        super().__init__(api, **kwargs)
+
+    def _get_page(self) -> None:
+        """
+        Get the scan results for the next target scan ID.
+        """
+        current_target_scan_id = self.target_scan_ids.pop()
+        self._log.debug(f"Getting the data for target ID: {current_target_scan_id}")
+        self.page = self._api.download_scan_report(current_target_scan_id)["findings"]
```

Comment:
As per the mapping document we need the finalized_at timestamp for checkpoint logic and scan_id and config_id as well for dashboards